### DISCUSSION TITLE
Dynamic Dialog templates

### DISCUSSION THREAD

**Discussion by AlexMonakhov** [primefaces/community#185] [PrimeVue] [answered]
Hey y'all, I started using PrimeVue and had issue with templates in dynamic dialog. I want to add container template in dialog like
```
templates: {
      container: markRaw(SomeContainerComponent)
    }
```
due to Documentation
[DynamicDialog uses the Dialog component internally, visit dialog for more information.](https://primevue.org/dynamicdialog/#customization)
and
[Dialog slots](https://primevue.org/dynamicdialog/#api.dialog.slots) (which mentioned in DynamicDialogAPI)

but dynamic dialog exports and as i understood supports only
 ```
export interface DynamicDialogTemplates {
    /**
     * Custom header template.
     */
    header?: any;
    /**
     * Custom footer template.
     */
    footer?: any;
}
```
Do we have any solution to solve this issue? Maybe some other ways to do it?
Thanks in advance.

**Comment by tugcekucukoglu** [comment, marked as answer]
Hey,

Dialog also has a default slot and the first parameter of the open function represents the default slot. Like in the below example ProductListDemo.

```
const dialogRef = dialog.open(ProductListDemo, {/*other options*/});
```

On the other hand, the container slot wraps the whole dialog (header, footer, and default slot). If the container slot is to be used, there is no need to use the default slot. A backward compatible way can be considered for DynamicDialog.
